components/style_traits/values/font.rs

```diff
@@ -0,0 +1,379 @@
+        // string (as lowercase) in the static atoms table. We don't have an
+        // API to do that yet though, so we do the simple thing for now.
+        let mut css_wide_keyword = false;
+        match_ignore_ascii_case! { &first_ident,
+            "serif" => return Ok(SingleFontFamily::Generic(atom!("serif"))),
+            "sans-serif" => return Ok(SingleFontFamily::Generic(atom!("sans-serif"))),
+            "cursive" => return Ok(SingleFontFamily::Generic(atom!("cursive"))),
+            "fantasy" => return Ok(SingleFontFamily::Generic(atom!("fantasy"))),
+            "monospace" => return Ok(SingleFontFamily::Generic(atom!("monospace"))),
+
+            #[cfg(feature = "gecko")]
+            "-moz-fixed" => return Ok(SingleFontFamily::Generic(atom!("-moz-fixed"))),
+
+            // https://drafts.csswg.org/css-fonts/#propdef-font-family
+            // "Font family names that happen to be the same as a keyword value
+            //  (`inherit`, `serif`, `sans-serif`, `monospace`, `fantasy`, and `cursive`)
+            //  must be quoted to prevent confusion with the keywords with the same names.
+            //  The keywords ‘initial’ and ‘default’ are reserved for future use
+            //  and must also be quoted when used as font names.
+            //  UAs must not consider these keywords as matching the <family-name> type."
+            "inherit" => css_wide_keyword = true,
+            "initial" => css_wide_keyword = true,
+            "unset" => css_wide_keyword = true,
+            "default" => css_wide_keyword = true,
+            _ => {}
+        }
+
+        let mut value = first_ident.as_ref().to_owned();
+        let mut serialization = String::new();
+        serialize_identifier(&first_ident, &mut serialization).unwrap();
+
+        // These keywords are not allowed by themselves.
+        // The only way this value can be valid with with another keyword.
+        if css_wide_keyword {
+            let ident = input.expect_ident()?;
+            value.push(' ');
+            value.push_str(&ident);
+            serialization.push(' ');
+            serialize_identifier(&ident, &mut serialization).unwrap();
+        }
+        while let Ok(ident) = input.try(|i| i.expect_ident_cloned()) {
+            value.push(' ');
+            value.push_str(&ident);
+            serialization.push(' ');
+            serialize_identifier(&ident, &mut serialization).unwrap();
+        }
+        Ok(SingleFontFamily::FamilyName(FamilyName {
+            name: Atom::from(value),
+            syntax: FamilyNameSyntax::Identifiers(serialization),
+        }))
+    }
+
+    #[cfg(feature = "gecko")]
+    /// Return the generic ID for a given generic font name
+    pub fn generic(name: &Atom) -> (structs::FontFamilyType, u8) {
+        use gecko_bindings::structs::FontFamilyType;
+        if *name == atom!("serif") {
+            (FontFamilyType::eFamily_serif,
+             structs::kGenericFont_serif)
```

Comment:
There's no way this builds, given `structs` is not defined here. This needs to build also with `./mach build-geckolib`.